### PR TITLE
[Project Model] Save severity in C# code analysis

### PR DESCRIPTION
**Pull request by ketttY**
Add handler for changed severity, update names from notificationPreference to severity for better consistency.

Fixes: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/952602

### CODE REVIEW

File: main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring.Options/UI/GridOptionPreviewControl.cs [hunk 1]
```diff
@@ -118,10 +118,31 @@ void CreateView ()
...
 			severityCellView.Editable = true;
 			severityCellView.ItemsField = notificationPreferencesField;
+			severityCellView.SelectionChanged += delegate (object sender, WidgetEventArgs e) {
+				var treeNavigator = store.GetNavigatorAt (treeView.CurrentEventRow);
+				if (treeNavigator == null)
+					return;
+				var item = treeNavigator.GetValue (itemField);
+				if (item == null)
+					return;
+				var oldSeverity = treeNavigator.GetValue (severityField);
```
Comment (mrward) [inline main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring.Options/UI/GridOptionPreviewControl.cs]: oldSeverity does not seem to be used. It looks like we do not need to get this value.

File: main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring.Options/UI/GridOptionPreviewControl.cs [hunk 2]
```diff
@@ -118,10 +117,30 @@ void CreateView ()
 			col.Expands = true;
 			treeView.Columns.Add (col);
 
-			var severityCellView = new ComboBoxCellView (notificationPreferenceField);
+			var severityCellView = new ComboBoxCellView (severityField);
 			severityCellView.Editable = true;
-			severityCellView.ItemsField = notificationPreferencesField;
+			severityCellView.ItemsField = severitiesField;
+			severityCellView.SelectionChanged += delegate (object sender, WidgetEventArgs e) {
```
Comment (jtorres) [inline main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring.Options/UI/GridOptionPreviewControl.cs]: don't we have to unsubscribe `SelectionChanged` when Dispose? // @Therzok

Comment (Therzok) [reply main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring.Options/UI/GridOptionPreviewControl.cs]: We should, but other handlers in the file do that anyway.

I plan to write a roslyn analyzer that errors out when we don't unsubscribe handlers.

**Comment by sgmunn** [comment]
@monojenkins backport release-8.3

**Comment by sgmunn** [comment]
test failures are unrelated.

**Pull request merged**